keps/sig-node/20200110-apparmor-ga.md

```diff
@@ -0,0 +1,390 @@
+enforce AppArmor, and prevents the kubelet from needing to resolve annotations &
+fields.
+
+If both AppArmor annotations _and_ fields are specified, the values MUST match.
+This will be enforced in API validation.
+
+#### Pod Update
+
+The AppArmor fields on a pod are immutable.
+
+The behavior on annotation update is currently ill-defined: the annotation
+update is allowed, but the new value will not be used until the container is
+restarted. There is no way to tell (from the API) what value a container is
+using.
+
+Therefore, AppArmor annotation updates will be ignored. This maintains backwards
+API compatibility (no tightening validation), and makes a small stabilizing
+change to behavior (new Kubelets will ignore the update).
+
+#### PodSecurityPolicy Creation
+
+_Similar policy to [Pod Creation](#pod-creation)_
+
+If no AppArmor annotations or fields are specified, no action is necessary.
+
+If _only_ AppArmor fields are specified, add the corresponding annotations. This
+ensures that the fields are enforced even in skewed high-availability
+environments. Since [we
+support](https://kubernetes.io/docs/setup/release/version-skew-policy/) up to 1
+minor release of version skew between the HA masters, this behavior will be
+removed in version N+1 (where N is the release with the upgraded AppArmor
+support).
+
+If _only_ AppArmor annotations are specified, copy the values into the
+corresponding fields. This smooths the migration for a future release that
+removes annotation support, and frees tools from needing to check annotations.
+
+If both AppArmor annotations _and_ fields are specified, the values MUST match.
+This will be enforced in API validation.
+
+#### PodSecurityPolicy Update
+
+If the old pod did not specify any AppArmor options (fields or annotations),
+follow the same rules as for creation.
+
+If only the fields or only the annotations are changed, take the new value and
+copy it to both the corresponding fields & annotations (even if one was
+previously unset).
+
+If both the fields _and_ annotations are changed, the new values MUST match.
+Enforced in API validation.
+
+#### PodSecurityPolicy Enforcement
+
+The PodSecurityPolicy admission controller must continue to check the PSP object
+for annotations if the fields are unset, in order to handle upgrade & version
+skew scenarios.
+
+When setting default profiles, PSP only needs to set the field. The API
+machinery will handle setting the annotation as necessary.
+
+When enforcing allowed profiles, the PSP should check BOTH the annotations and
+fields. In most cases, they should be consistent. On pod update, the AppArmor
+annotations may differ from the fields. In that case, the PSP enforcement should
+check both values as the effective value depends on the node version running the
+pod.
+
+#### PodTemplates
+
+PodTemplates (e.g. ReplaceSets, Deployments, StatefulSets, etc.) will be
+ignored. The field/annotation resolution will happen on template instantiation.
```

Comment:
Add the warning here too (see seccomp proposal)

Reply:
Added as suggested.